src/Nethermind/Nethermind.Evm/Tracing/GethStyle/GethLikeTxFileTracer.cs

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Nethermind.Evm.Tracing.GethStyle;
+
+public class GethLikeTxFileTracer : GethLikeTxTracer<GethTxFileTraceEntry>
+{
+    private readonly Action<GethTxFileTraceEntry> _dumpCallback;
+    private long? _startGas;
+
+    public GethLikeTxFileTracer(Action<GethTxFileTraceEntry> dumpCallback, GethTraceOptions options) : base(options)
+    {
+        _dumpCallback = dumpCallback ?? throw new ArgumentNullException(nameof(dumpCallback));
+
+        IsTracingMemory = true;
+        IsTracingOpLevelStorage = false;
+        IsTracingRefunds = true;
+    }
+
+    public override GethLikeTxTrace BuildResult()
+    {
+        var trace = base.BuildResult();
+
+        if (_startGas.HasValue)
+            trace.Gas = _startGas.Value - CurrentTraceEntry.Gas;
+
+        return trace;
+    }
+
+    public override void ReportRefund(long refund) => CurrentTraceEntry.Refund = refund;
+
+    protected override void AddTraceEntry(GethTxFileTraceEntry entry)
```

Comment:
In file format GethTxTraceEntry and it's collections could be pooled at the tracer level, reducing allocations even more!

Reply:
I'm not sure I follow. Could you please elaborate?

Reply:
GethTxFileTraceEntry are directly serialized to JSON, right? If so we can have a Pool of them and instead of creating and then  dropping them, we can clear them and return to the pool. When creating try getting them from pool. Would reduce allocations a lot.

Reply:
Now it simply reuses the same instance. No pooling.